src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionPal.Windows.cs

```diff
@@ -201,6 +207,10 @@ private static unsafe void ProcessResult(SocketError errorCode, GetAddrInfoExCon
                             AddressList = addresses
                         });
                 }
```

Comment:
Consider changing the below to be something instead like:
```C#
else
{
    Exception e = errorCode == (SocketError)Interop.Winsock.WSA_E_CANCELLED && cancellationToken.IsCancellationRequested ? (Exception)
        new OperationCanceledException(cancellationToken) :
        new SocketException((int)errorCode);
    state.SetResult(ExceptionDispatchInfo.SetCurrentStackTrace(e));
}
```
to avoid the duplication, and have one path for success and one path for failure.